clinguin_ui/ooasp_backend.py

```diff
@@ -0,0 +1,106 @@
+from clinguin.server.application.backends import ClingraphBackend
+from ooasp.smart_ooasp import SmartOOASPSolver
+from clingo import Function, Control, parse_term
+
+
+SAVE_FILE = "ooasp_configuration.lp"
+
+
+ASSOCIATION_SPECIALIZATIONS = [
+    "rack_framesS",
+    "rack_framesD",
+    "element_modules1",
+    "element_modules2",
+    "element_modules3",
+    "element_modules4",
+]
+
+
+class OOASPBackend(ClingraphBackend):
+
+    def _init_ctl(self):
+        super()._init_ctl()
+        self.smart_solver = SmartOOASPSolver(
+            initial_objects=[],
+            ctl=self._ctl,
+            verbose=True,
+            associations_with_priority=ASSOCIATION_SPECIALIZATIONS,
+            smart_generation_functions=[
+                "object_needed",
+                "global_lb",
+                "global_ub",
+                "association_needed",
+            ],
+        )
+        self.smart_solver.load_base()
+        # self.smart_solver.create_initial_objects()
```

Comment:
Please don't include commented-out code

Reply:
I will recheck all the files for commented out code.